core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/action/ImportDictionaryServlet.java

```diff
@@ -0,0 +1,132 @@
+                    response.setStatus(400, error);
+                    return;
+                }
+                CSVParser csvParser = new CSVParser(reader, format);
+                Map<String, Integer> headers = csvParser.getHeaderMap();
+
+                if (!headers.containsKey("KEY") || headers.get("KEY") != 0) {
+                    String error = "Invalid CSV file. The first column must be 'KEY'. The Delimiter should be ',' or ';'.";
+                    LOG.error(error);
+                    response.setStatus(400, error);
+                    return;
+                }
+
+                headers.remove("KEY");
+                for (String language : headers.keySet()) {
+                    languages.add(language);
+                    translations.add(new ArrayList<>());
+                }
+
+                for (CSVRecord record : csvParser) {
+                    if (record.size() != languages.size() + 1) {  // +1 for the 'KEY' column.
+                        LOG.warn("Ignoring row with incorrect number of translations: " + record);
+                        continue;
+                    }
+
+                    String label = record.get("KEY");
+                    KEYs.add(label);
+
+                    for (String language : languages) {
+                        int index = languages.indexOf(language);
+                        String translation = record.get(language);
+                        translations.get(index).add(translation);
+
+                        Resource languageResource = resourceResolver.getResource(path + "/" + language);
+                        if (languageResource == null) {
+                            Map<String, Object> newFolderProperties = new HashMap<>();
+                            newFolderProperties.put("jcr:primaryType", "sling:Folder");
+                            newFolderProperties.put("jcr:language", language);
+                            newFolderProperties.put("jcr:basename", language);
+                            newFolderProperties.put("sling:resourceType", "sling:Folder");
+                            newFolderProperties.put("jcr:mixinTypes", new String[]{"mix:language"});
+                            languageResource = resourceResolver.create(resourceResolver.getResource(path), language, newFolderProperties);
+                        }
+
+                        Map<String, Object> newNodeProperties = new HashMap<>();
+                        newNodeProperties.put("jcr:primaryType", "nt:unstructured");
+                        String newNodeName = label;
+                        Resource labelResource = resourceResolver.getResource(path + "/" + language + "/" + newNodeName);
+                        if (labelResource == null) {
+                            labelResource = resourceResolver.create(languageResource, newNodeName, newNodeProperties);
+                        }
+                        ModifiableValueMap mvm = labelResource.adaptTo(ModifiableValueMap.class);
```

Comment:
don't abbreviate variable names because it's nog very readable

Reply:
modifiableValueMap